File: conformance/tests/multi-method/multi-method.runar.py
from runar import SmartContract, PubKey, Sig, Bigint, public, assert_, check_sig

class MultiMethod(SmartContract):
    owner: PubKey
    backup: PubKey

    def __init__(self, owner: PubKey, backup: PubKey):
        super().__init__(owner, backup)
        self.owner = owner
        self.backup = backup

    def _compute_threshold(self, a: Bigint, b: Bigint) -> Bigint:
        return a * b + 1

    @public
    def spend_with_owner(self, sig: Sig, amount: Bigint):
        threshold: Bigint = self._compute_threshold(amount, 2)
        assert_(threshold > 10)
        assert_(check_sig(sig, self.owner))

    @public
    def spend_with_backup(self, sig: Sig):
        assert_(check_sig(sig, self.backup))

File: conformance/tests/multi-method/multi-method.runar.rb
require 'runar'

class MultiMethod < Runar::SmartContract
  prop :owner, PubKey
  prop :backup, PubKey

  def initialize(owner, backup)
    super(owner, backup)
    @owner = owner
    @backup = backup
  end

  params a: Bigint, b: Bigint
  def compute_threshold(a, b)
    return a * b + 1
  end

  runar_public sig: Sig, amount: Bigint
  def spend_with_owner(sig, amount)
    threshold = compute_threshold(amount, 2)
    assert threshold > 10
    assert check_sig(sig, @owner)
  end

  runar_public sig: Sig
  def spend_with_backup(sig)
    assert check_sig(sig, @backup)
  end
end
